apps/api/src/services/data-provider.service.ts

```diff
@@ -184,4 +185,22 @@ export class DataProviderService implements DataProviderInterface {
 
     return dataOfYahoo;
   }
+
+  public async search(aSymbol: string) {
+    let results: { items: LookupItem[] } = { items: [] };
+
+    switch (this.configurationService.get('DATA_SOURCES')[0]) {
```

Comment:
I would propose to extract the switch case which is determining the `dataProviderInterface` to use to a separate method and then call the `.search(...)` in here, outside the switch case. This will increase the re-usability once we have new methods.

Reply:
I don't understand 😳 Can you give an example?

Reply:
Something like this:
```typescript
    public async search(aSymbol: string) {
        return this.getDataProvider().search(aSymbol);
    }

    private getDataProvider(): DataProviderInterface {
        switch (this.configurationService.get('DATA_SOURCES')[0]) {
            case DataSource.ALPHA_VANTAGE:
                return this.alphaVantageService;
                break;
            case DataSource.YAHOO:
                return this.yahooFinanceService;
                break;
            default:
                throw new Error('No data provider has been found.');
        }
    }
```

Reply:
Okay, that looks nice. Thanks @vzickner! 😎